### PR TITLE
LibJS: Implement almost all Math functions

### PR DESCRIPTION
**Pull request by Cleverking2003**
In this PR I have implemented almost all Math functions (except for Math.imul() because it should treat arguments as C-type ints). However, some of the functions fail tests because of #4543

### CODE REVIEW

File: Libraries/LibJS/Runtime/MathObject.cpp [hunk 1]
```diff
@@ -357,4 +366,138 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::cbrt)
...
+    return Value(::sqrt(hypot.as_double()));
+}
+
+JS_DEFINE_NATIVE_FUNCTION(MathObject::log)
+{
+    auto number = vm.argument(0).to_number(global_object);
+    if (vm.exception())
+        return {};
+    if (number.as_double() < 0)
+        return JS::js_nan();
```
Comment (linusg) [inline Libraries/LibJS/Runtime/MathObject.cpp:445]: We're in the JS namespace already :)

```suggestion
        return js_nan();
```

File: Libraries/LibJS/Runtime/MathObject.cpp [hunk 2]
```diff
@@ -357,4 +366,138 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::cbrt)
...
+    return Value(::log(number.as_double()));
+}
+
+JS_DEFINE_NATIVE_FUNCTION(MathObject::log2)
+{
+    auto number = vm.argument(0).to_number(global_object);
+    if (vm.exception())
+        return {};
+    if (number.as_double() < 0)
+        return JS::js_nan();
```
Comment (linusg) [inline Libraries/LibJS/Runtime/MathObject.cpp:455]:
```suggestion
        return js_nan();
```

File: Libraries/LibJS/Runtime/MathObject.cpp [hunk 3]
```diff
@@ -357,4 +366,138 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::cbrt)
...
+    return Value(::log2(number.as_double()));
+}
+
+JS_DEFINE_NATIVE_FUNCTION(MathObject::log10)
+{
+    auto number = vm.argument(0).to_number(global_object);
+    if (vm.exception())
+        return {};
+    if (number.as_double() < 0)
+        return JS::js_nan();
```
Comment (linusg) [inline Libraries/LibJS/Runtime/MathObject.cpp:465]:
```suggestion
        return js_nan();
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Libraries/LibJS/Runtime/MathObject.cpp [hunk 4]
```diff
@@ -357,4 +366,138 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::cbrt)
     return Value(::cbrt(number.as_double()));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(MathObject::atan2)
+{
+    auto y = vm.argument(0).to_number(global_object), x = vm.argument(1).to_number(global_object);
+    auto pi_4 = M_PI_2 / 2, three_pi_4 = pi_4 + M_PI_2;
```
Comment (linusg) [inline Libraries/LibJS/Runtime/MathObject.cpp:372]: declaring multiple variables at once is uncommon for this project's coding style, I think.

```suggestion
    auto pi_4 = M_PI_2 / 2;
    auto three_pi_4 = pi_4 + M_PI_2;
```

**Review by linusg — COMMENTED** [review]
LGTM, thanks for working on this! Please squash the third commit into the first as per the [code submission policy](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#code-submission-policy).

**Comment by Cleverking2003** [comment]
Argh, I messed up squashing and now I will need to rewrite tests :(

**Comment by Lubrsi** [comment]
You can recover them here: https://github.com/SerenityOS/serenity/commit/8b6775dffb1fe83db444b3e1968179251ad86c45

**Comment by Lubrsi** [comment]
Just need to re-add new lines at the end of the tests and should be good to go.

**Comment by Cleverking2003** [comment]
Ok, everything done

**Comment by Lubrsi** [comment]
:+1: Just have to wait for final say from Andreas

**Pull request merged**